services-bytebuf-codec/src/test/java/io/scalecube/services/transport/api/DefaultHeadersCodecTest.java

```diff
@@ -0,0 +1,46 @@
+package io.scalecube.services.transport.api;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class DefaultHeadersCodecTest {
+
+  private HeadersCodec codec = new DefaultHeadersCodec();
+
+  @ParameterizedTest
+  @MethodSource("provider")
+  void test(Map<String, String> headers) throws IOException {
+    Map<String, String> decoded = writeAndRead(headers);
+    assertEquals(headers, decoded);
+  }
+
+  static Stream<Map<String, String>> provider() {
+    Map<String, String> sampleMap = new HashMap<>();
+    sampleMap.put("header", "value");
+    sampleMap.put("test", "3");
```

Comment:
What happens with an empty string as a value?

Reply:
i will add this cases in test